modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set(pdl_dir             ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1)
+set(pdl_drv_dir         ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1/drivers)
+set(pdl_dev_cat1a_dir   ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-pdl-cat1/devices/COMPONENT_CAT1A)
+
+zephyr_compile_definitions(${CONFIG_SOC_PART_NUMBER})
+
+# Add mtb-pdl-cat1
+zephyr_include_directories(${pdl_drv_dir}/include)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/include)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/include/ip)
+zephyr_include_directories(${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB)
+
+zephyr_library_sources(${pdl_dev_cat1a_dir}/source/cy_device.c)
+zephyr_library_sources(${pdl_drv_dir}/source/TOOLCHAIN_GCC_ARM/cy_syslib_gcc.S)
+
+zephyr_library_sources_ifdef(CONFIG_CPU_CORTEX_M4
+  ${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB/COMPONENT_CM4/system_psoc6_cm4.c)
+zephyr_library_sources_ifdef(CONFIG_CPU_CORTEX_M0PLUS
+  ${pdl_dev_cat1a_dir}/templates/COMPONENT_MTB/COMPONENT_CM0P/system_psoc6_cm0plus.c)
+
+# Peripheral drivers
+zephyr_library_sources_ifdef(CONFIG_SERIAL ${pdl_drv_dir}/source/cy_scb_uart.c)
+zephyr_library_sources_ifdef(CONFIG_ADC ${pdl_drv_dir}/source/cy_sar.c)
+zephyr_library_sources_ifdef(CONFIG_I2C ${pdl_drv_dir}/source/cy_scb_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_SPI ${pdl_drv_dir}/source/cy_scb_spi.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER ${pdl_drv_dir}/source/cy_tcpwm_counter.c)
+zephyr_library_sources_ifdef(CONFIG_PWM ${pdl_drv_dir}/source/cy_tcpwm_pwm.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6 ${pdl_drv_dir}/source/cy_sysint.c)
+
+if(CONFIG_SERIAL OR CONFIG_I2C OR CONFIG_SPI)
+  zephyr_library_sources(${pdl_drv_dir}/source/cy_scb_common.c)
+endif()
+
+if(CONFIG_ADC)
+  zephyr_library_sources(${pdl_drv_dir}/source/cy_sysanalog.c)
+endif()
+
+# Common part
+zephyr_library_sources(${pdl_drv_dir}/source/cy_flash.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_gpio.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_ipc_drv.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_ipc_pipe.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_ipc_sema.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_prot.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_sd_host.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_sysclk.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_syslib.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_syspm.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_systick.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_trigmux.c)
+zephyr_library_sources(${pdl_drv_dir}/source/cy_wdt.c)
```

Comment:
some of these sources seems to be optional, e.g. `cy_gpio.c` should only be compiled if the Cypress GPIO driver selects to use HAL module.

Reply:
cy_gpio.c is used in almost all hal modules in cyhal_pin structure. so i added it to common part (without conditions).

we can add this under USE_INFINEON_GPIO, and then enable mandatory it in SOC_FAMILY_PSOC6
but looks likes if we go with that way, we need to add USE_INFINEON_XXX for all common part.

I would like to file internal ticket and update in scope of another PR (e.g. zephyr/pull/44211). Please provide your thoughts.